src/views/Layout.jsx

```diff
@@ -26,6 +26,9 @@ export function Layout() {
 				<NavLink to="/summary" className="Nav-link">
 					Summary
 				</NavLink>
+				<div
+					className={`Nav-link__shadow Nav-link__shadow_${activeComponent}`}
+				></div>
```

Comment:
Suggestion: instead of setting state for every route component, you could use `location.pathname` from React Router's [useLocation](https://reactrouter.com/docs/en/v6/hooks/use-location) to get the current path name and render the active component from that string.

Reply:
Ahhh I was looking for a solution there! Let me try!

Reply:
@xynree Refactored it with `window.location.pathname` in `Layout.js`. Put it in a separate react component so it doesn't re-render the entire Layout component and all its child component. Not exactly sure if there's a better way. What do you think?